ionoscloud/resource_application_loadbalancer.go

```diff
@@ -147,6 +159,25 @@ func resourceApplicationLoadBalancerCreate(ctx context.Context, d *schema.Resour
 		diags := diag.FromErr(errState)
 		return diags
 	}
+	if flowLogs, ok := d.GetOk("flowlog"); ok {
+		fw := cloudapiflowlog.Service{
+			D:      d,
+			Client: client,
+		}
+		if flowLogList, ok := flowLogs.([]any); ok {
+			for _, flowLogData := range flowLogList {
+				if flowLogMap, ok := flowLogData.(map[string]interface{}); ok {
+					flowLog := cloudapiflowlog.GetFlowlogFromMap(flowLogMap)
+					err := fw.CreateOrPatchForALB(ctx, dcId, d.Id(), "", flowLog)
+					if err != nil {
+						d.SetId("")
```

Comment:
Should use d.Set("flowlog",nil) ?
Otherwise ALB will be deleted from state file even though it has been successfully created previously.

Reply:
good catch!

Reply:
there is a discussion if we shouldn't fail the entire resource though if flowlog creation fails when we create the entire resource.

Reply:
It would make sense to fail the entire load balancer creation, since you might want to capture all flow logs from the moment the alb resource is created.
Having only the flowlog creation fail, would it mean that if you add it to the lb later on in an update, you would miss all captures prior to that point?

Reply:
That is correct. Failing the entire resource would still mean that you get the alb created in the backend unless we implement rollback.

Reply:
implemented fallback